contracts/ModuleBase.sol

```diff
@@ -30,27 +33,40 @@ abstract contract ModuleBase is IModuleBase, UUPSUpgradeable, ERC165 {
         public
         view
         virtual
-        override
+        override( ERC165, IModuleBase)
         returns (bool)
     {
         return
             interfaceId == type(IModuleBase).interfaceId ||
             super.supportsInterface(interfaceId);
     }
 
+    function currentImpl() public view returns(address) {
+        return _getImplementation();
+    }
+
+    function isImplCurrent() public view returns(address uptodateImpl, bool isCurrent) {
+        uptodateImpl =  IModuleFactory(moduleFactoryBase).currentVersionInfo().impl;
+        uptodateImpl == currentImpl() ? isCurrent = true : isCurrent = false;
+    }
```

Comment:
Does this need to be in here?

The concept of versioning is more nuanced than simply "are we at the latest version or not" (e.g. projects might have very good reasons to not want to upgrade between major version of a package, but still want bugfixes for the version they're on).

I'm thinking that this is more of a frontend detail, than something that needs to be resolved on-chain.

Reply:
agreed. my thought here was for this to be read only function used by the frontend to quickly query the factory state

Reply:
yeahhh, but my bigger point is that "current" means different things to different projects.

imagine a library that has two major versions in the wild, 1.x.x and 2.x.x

the maintainers might still be shipping bugfixes to the 1.x.x release branch, while also adding features and fixing bugs on the 2.x.x branch

some Big Project might be happy using the 1.x.x branch, because version 2.x.x removed some functionality that Big Project needs.

So according to Big Project, what is "the current version"? According to them, it's the highest bugfix version on the major version starting with "1".

Again, let's not have the contracts assume these things.

Oh! another point. The way the code is implemented now, you're assuming that the "latest version" is always the most recently pushed item into the version array.

But what if a module dev pushes a new major version, but then later pushes a bugfix version to their previous major version release? Then your assumption breaks probably.

Point of "semantic versioning" is to address the problems that happen when "releases" and "versions" are assumed to be linear.